src/SiSenseWriter.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\SiSenseWriter;
+
+use Keboola\Component\UserException;
+use Psr\Log\LoggerInterface;
+
+class SiSenseWriter
+{
+    private string $dataDir;
+
+    private Api $api;
+
+    private Config $config;
+
+    private LoggerInterface $logger;
+
+    public function __construct(string $dataDir, Config $config, Api $api, LoggerInterface $logger)
+    {
+        $this->dataDir = $dataDir;
+        $this->api = $api;
+        $this->config = $config;
+        $this->logger = $logger;
+
+        $this->api->login();
+    }
+
+    public function execute(): void
+    {
+        $csvFile = $this->api->uploadFile(
+            sprintf('%s/in/tables/', $this->dataDir),
+            sprintf('%s.csv', $this->config->getTableId())
+        );
+
+        $datamodel = $this->createAndGetDatamodel();
+
+        $dataset = $this->createAndGetDataset($datamodel, $csvFile);
+
+        $table = $this->createAndGetTable($datamodel, $dataset);
+
+        if ($this->config->getRelationships()) {
+            $this->createRelationships($datamodel, $dataset, $table);
+        }
+
+        $this->buildData($datamodel);
+    }
+
+    private function createAndGetDatamodel(): array
+    {
+        $datamodel = $this->api->checkDatamodelExists($this->config->getDatamodelName());
+        if (is_null($datamodel)) {
+            $this->logger->info(sprintf('Create new datamodel "%s"', $this->config->getDatamodelName()));
+            $datamodel = $this->api->createDatamodel($this->config->getDatamodelName());
+        } else {
+            $this->logger->info(sprintf('Update existing datamodel "%s"', $datamodel['title']));
+        }
+        return $datamodel;
+    }
+
+    private function createAndGetDataset(array $datamodel, string $csvFile): array
+    {
+        $dataset = $this->api->checkDatasetExists(
```

Comment:
Bolo by lepsie ak by ti trieda `Api` vracala objekty, ... lebo takto nemas definovane typy, a musis si pamatat indexy ... ak by si mal objekty (`ValueObject`), tak das iba `$dataset->getId()`, priamo ti to navrhne IDE-cko a lepsie sa s tym pracuje.

Reply:
jj máš pravdu - hotovo